### PR TITLE
Lvnodcdxi 29

### PR DESCRIPTION
**Pull request by OstapDutkevych**
Added drag&drop (Upload music)

### CODE REVIEW

File: src/client/components/dragAndDrop/drag-and-drop.css [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.container:hover {
+    color: rgb(255, 255, 255);
```
Comment (StanislavFedyna) [inline src/client/components/dragAndDrop/drag-and-drop.css]: Don't use RGB, use please standart form of color (#*******)

File: src/client/components/dragAndDrop/drag-and-drop.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Pizzicato from 'pizzicato';
+import Dropzone from 'react-dropzone';
+import { createAudioData, playPauseSoundFromFile } from '../../actions/audioActions';
+
+import './drag-and-drop.css';
```
Comment (StanislavFedyna) [inline src/client/components/dragAndDrop/drag-and-drop.js]: Change name for of css file, like dragAndDrop.css

File: src/client/components/dragAndDrop/drag-and-drop.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      }, () => this.createSoundInfoInState(sound, file));
+    };
+
+  createSoundInfoInState = (sound, file) => {
+    const { audioData: { analyser }, createAudioDataAsProp } = this.props;
+    sound.connect(analyser);
+    createAudioDataAsProp({
+      sound,
+      trackName: file.name,
+      trackType: file.type,
```
Comment (StanislavFedyna) [inline src/client/components/dragAndDrop/drag-and-drop.js]: We already don't use trackName and trackType, something goes wrong with your merge :(

File: src/client/components/equalizer/canvasEqualizer/canvasEqualizer.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+/* .graphicCanvas{
```
Comment (mr-redFox) [inline src/client/components/equalizer/canvasEqualizer/canvasEqualizer.css]: never use comments for code, only for describe functionality

File: src/client/components/dragAndDrop/drag-and-drop.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      trackSize: file.size,
+    });
+  };
+
+  render() {
+    const maxSize = 1000000000;
+    return (
+      <div className="container text-center mt-5">
```
Comment (StanislavFedyna) [inline src/client/components/dragAndDrop/drag-and-drop.js]: I think it's not a good idea add a name like text-center and mt-5

File: src/client/components/dragAndDrop/drag-and-drop.js (hunk 2 above)
Comment (StanislavFedyna) [reply src/client/components/dragAndDrop/drag-and-drop.js]: And seems problem you have with naming of js file

File: src/client/components/dragAndDrop/drag-and-drop.js (hunk 5 above)
Comment (OstapDutkevych) [reply src/client/components/dragAndDrop/drag-and-drop.js]: @StanislavFedyna MARGIN-TOP: 5px; you understand me?

File: src/client/components/equalizer/canvasEqualizer/canvasEqualizer.css (hunk 4 above)
Comment (OstapDutkevych) [reply src/client/components/equalizer/canvasEqualizer/canvasEqualizer.css]: done

File: src/client/components/dragAndDrop/drag-and-drop.js (hunk 3 above)
Comment (OstapDutkevych) [reply src/client/components/dragAndDrop/drag-and-drop.js]: done

File: src/client/components/dragAndDrop/drag-and-drop.js (hunk 2 above)
Comment (OstapDutkevych) [reply src/client/components/dragAndDrop/drag-and-drop.js]: done

File: src/client/components/dragAndDrop/drag-and-drop.css (hunk 1 above)
Comment (OstapDutkevych) [reply src/client/components/dragAndDrop/drag-and-drop.css]: done

File: src/client/components/dragAndDrop/drag-and-drop.js (hunk 5 above)
Comment (StanislavFedyna) [reply src/client/components/dragAndDrop/drag-and-drop.js]: change this 💩
We don't use a style like in bootstrap or something like that
If you need margin, you can use margin-top or text-align: center for centering of text

File: src/client/components/dragAndDrop/dragAndDrop.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import Pizzicato from 'pizzicato';
+import Dropzone from 'react-dropzone';
+import { createAudioData, playPauseSoundFromFile } from '../../actions/audioActions';
+
+import './dragAndDrop.css';
+
+
+class DragAndDrop extends Component {
+    onDrop = (acceptedFiles) => {
+      const file = acceptedFiles[0];
```
Comment (tonypntsrn) [inline src/client/components/dragAndDrop/dragAndDrop.js]: You could use destruction here const [file] = acceptedFiles;

File: src/client/components/equalizer/equalizer.js [hunk 7]
```diff
@@ -189,8 +187,38 @@ class Equalizer extends Component {
       widthCanvas,
       heightCanvas,
       trackName,
+      sound,
+      startMuteState,
     } = this.props.audioData;
 
+    const hideOrShowCanvas = sound || startMuteState ? (
```
Comment (tonypntsrn) [inline src/client/components/equalizer/equalizer.js]: No need to repeat code, you could use ternary on styles you want to change

File: src/client/components/equalizer/equalizer.js [hunk 8]
```diff
@@ -189,8 +187,38 @@ class Equalizer extends Component {
...
+      <div>
+        <div style={{ display: 'block' }}>
+          <Graphicequaliser
```
Comment (tonypntsrn) [inline src/client/components/equalizer/equalizer.js]: Please, rename this component using camel style

File: src/client/components/dragAndDrop/dragAndDrop.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+          minSize={0}
+          maxSize={maxSize}
+        >
+          {({
+            getRootProps, getInputProps, isDragActive, isDragReject,
+          }) => (
+            <div {...getRootProps()} className="containerForInput">
+              <input {...getInputProps()} />
+              {!isDragActive && 'Click here or drop a file to upload!'}
+              {isDragActive && !isDragReject && "Drop it like it's hot!"}
```
Comment (NadiiaChorna) [inline src/client/components/dragAndDrop/dragAndDrop.js]: Сhoose one way to write quotes. I think it's better to use only single quotation marks.

Comment (OstapDutkevych) [reply src/client/components/dragAndDrop/dragAndDrop.js]: okey)
done

File: src/client/components/equalizer/equalizer.js (hunk 8 above)
Comment (OstapDutkevych) [reply src/client/components/equalizer/equalizer.js]: done

File: src/client/components/equalizer/equalizer.js (hunk 7 above)
Comment (OstapDutkevych) [reply src/client/components/equalizer/equalizer.js]: done

File: src/client/components/dragAndDrop/dragAndDrop.js (hunk 6 above)
Comment (OstapDutkevych) [reply src/client/components/dragAndDrop/dragAndDrop.js]: done

File: src/client/components/dragAndDrop/drag-and-drop.js (hunk 5 above)
Comment (OstapDutkevych) [reply src/client/components/dragAndDrop/drag-and-drop.js]: done

**Review by StanislavFedyna — DISMISSED** [review]
Good job, малий)

File: src/client/components/equalizer/equalizer.js [hunk 10]
```diff
@@ -219,16 +224,21 @@ class Equalizer extends Component {
 
     return (
       <div className="graphicEqualizer">
-        <Graphicequaliser
-          width={widthCanvas}
-          height={heightCanvas}
-          getCanvasEl={setCanvasToState}
-        />
+        <div style={{ display: display2 }}>
```
Comment (tonypntsrn) [inline src/client/components/equalizer/equalizer.js]: Those ternarys could be made here, no need to create variables for that

Comment (OstapDutkevych) [reply src/client/components/equalizer/equalizer.js]: @tonypntsrn done

**Review by tonypntsrn — APPROVED** [review]

**Review by StanislavFedyna — APPROVED** [review]

**Pull request merged**